### PR TITLE
add shuffle to dataset

### PR DESCRIPTION
**Pull request by rickycao-qy**
The data in dataset should be shuffled to achieve better performance

### CODE REVIEW

File: packages/core/src/utils/public.ts [hunk 1]
```diff
@@ -290,3 +290,15 @@ export function getOsInfo() {
     });
   });
 }
+
+/**
+ * shuffle array
+ * @param array
```
Comment (yorkie) [inline packages/core/src/utils/public.ts:296]: Could we have more documentation for this method?

File: packages/core/src/utils/public.ts [hunk 2]
```diff
@@ -290,3 +290,15 @@ export function getOsInfo() {
...
+ */
+export function shuffle(array: any[]) {
```
Comment (yorkie) [inline packages/core/src/utils/public.ts:298]: Need a return value and unit tests for this function :)

File: packages/core/src/utils/public.ts (hunk 1 above)
Comment (rickycao-qy) [reply packages/core/src/utils/public.ts]: done

File: packages/core/src/utils/public.ts (hunk 2 above)
Comment (rickycao-qy) [reply packages/core/src/utils/public.ts:298]: shuffle will just operate original array (just like python's built-in shuffle function). I have added unit tests

File: packages/core/src/utils/public.ts [hunk 3]
```diff
@@ -290,3 +290,15 @@ export function getOsInfo() {
...
+/**
+ * Shuffles array in place. ES6 version. This method is based on Fisher-Yates shuffle algorithm
+ * @param array An array containing the items.
+ */
+export function shuffle(array: any[]) {
+  for (let i = array.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [ array[i], array[j] ] = [ array[j], array[i] ];
+  }
+  return array;
```
Comment (yorkie) [inline packages/core/src/utils/public.ts:303]: @rickycao-qy as you commented at https://github.com/alibaba/pipcook/pull/255#discussion_r443347354, we need to remove this return-statement to keep consistent.

Comment (rickycao-qy) [reply packages/core/src/utils/public.ts]: done

**Pull request merged**